### ISSUE TITLE
feature request: raise warning when enum values are duplicated

### ISSUE THREAD

**Issue by mboughaba** [palantir/tslint#3652]
### Bug Report

- __TSLint version__:
λ tslint --version
5.9.1
- __TypeScript version__:
λ tsc --version
Version 2.6.2
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```ts
    enum Color {
        Red = 5, // 5
        Green = 4, // 4
        Blue // 5!!
    }
```

#### Actual behavior
The code above is perfectly valid but can lead to bugs.

#### Expected behavior
Linter should raise a warning that enum values are duplicated.

#### Note
This is perfect Typescript code and I tried as well in C++, this is why I think can be fixed by a linter.
I will be happy to work on a PI if this fine.

**Comment by edwardboyle** [comment]
I will second this. 

Reason: 
We sometimes create complex enums instead of auto generated numbers (or strings now). I just found myself examining my enum very closely and thought, this has got to be in tslint. To my surprise, no. In my opinion this is a real lint, not just a neato. We have lots of neatos.

**Comment by vylesk** [comment]
I've just come across a developer maintained enum in a project I'm working on that has the same integer assignment for 2 enum values. I'm surprised to see after having to fix all the code styling warnings there's no check for this, which will actually cause problems in our system.

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
